backend/src/main/java/de/armbrust/planz/amazonapi/AmazonApiLogic.java

```diff
@@ -0,0 +1,32 @@
+package de.armbrust.planz.amazonapi;
+
+import com.amazon.sellingpartner.api.ReportsApi;
+import com.amazon.sellingpartner.model.ReportDocument;
+import com.amazon.spapi.documents.DownloadBundle;
+import de.armbrust.planz.model.Product;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+public class AmazonApiLogic {
+
+    public final ReportParsingService reportParsingService;
+    public final DownloadAndDecryptService downloadAndDecryptService;
+    public final ReportsApiService reportsApiService;
+
+    public AmazonApiLogic(ReportParsingService reportParsingService, DownloadAndDecryptService downloadAndDecryptService, ReportsApiService reportsApiService) {
+        this.reportParsingService = reportParsingService;
+        this.downloadAndDecryptService = downloadAndDecryptService;
+        this.reportsApiService = reportsApiService;
+    }
+
+    public List<Product> getProductsFromApiReport() {
+        ReportsApi reportsApi = reportsApiService.getReportsApi();
+        String reportsId = reportsApiService.createReportAndGetReportID(reportsApi);
```

Comment:
you get a value from reportsApiService and directly return this value into this class (see comment inside reportsApiService)